src/Illuminate/Database/DetectsLostConnections.php

```diff
@@ -40,6 +40,8 @@ protected function causedByLostConnection(Throwable $e)
             'Communication link failure',
             'connection is no longer usable',
             'Login timeout expired',
+            'Connection refused',
+            'running with the --read-only option so it cannot execute this statement',
```

Comment:
How is this a problem that can be solved by a reconnect?

Reply:
We are using read and write endpoints for the database connection in database.php

```
'mysql' => [
            'driver'    => 'mysql',
            'read'      => [
                'host' => env('DB_HOST_READ', 'localhost'),
            ],
            'write'     => [
                'host' => env('DB_HOST_WRITE', 'localhost'),
            ],
            'sticky'    => true,
            'database'  => env('DB_DATABASE', 'forge'),
            'username'  => env('DB_USERNAME', 'forge'),
            'password'  => env('DB_PASSWORD', ''),
            'charset'   => 'utf8',
            'collation' => 'utf8_unicode_ci',
            'prefix'    => '',
            'strict'    => false,
            'engine'    => null,
        ],
```

Laravel application is connected to writer endpoint, but then when Aurora fails over, that endpoint is now a reader, but laravel is trying to perform writes on it.